### ISSUE TITLE
Not working with Unity 5.10.3

### ISSUE THREAD

**Issue by a-jackson** [astaerk/AutoRegistration#8]
We've been using this for ages on an old version still running Microsoft.Practices.EnterpriseLibrary.Unity and are trying to upgrade Unity. 

I was happy to see this package updated as well however it's not working with the latest versions. 
This seems to be because you've built the project against Unity 5.1.2 which uses Unity.Abstractions 2.1. The latest version of Unity is using Unity.Abstractions 4.1.3 and it won't let us use Auto Registration which is depending on 2.1 with 4.1.3. There's appears to be breaking changes between these versions.

Would it be possible to release an update that is built on the newer version of Unity?

Thanks

**Comment by astaerk** [comment]
Wow, sorry for not answering for so long.
Can anyone tell me how I can get github to send me a mail when someone opens an issue in my own projects? I thought it would be that way, but obviously it isn't.

I'm going to check that.

**Comment by astaerk** [comment]
So, according to the unit tests it works up to version 5.8.x of Unity.
With 5.9.0 of Unity there were breaking changes.

I have a compilable version but I have a problem with registering open generics starting with that version. I need to have another look at that.
After that I'm very curious if there are more breaking changes in Unity beyond version 5.9.0.

**Comment by astaerk** [comment]
According to the unit tests I got it working now. So I'm very positive to have a release soon.
I'd like to do some manual testing before releasing it.

**Comment by a-jackson** [comment]
Hi - Thanks for looking at this. We had to shelve our Unity upgrade plans for now for this and a few other reasons so we've not been waiting on it. We will have to get to :) it at some point soon though so this will be a great help.

**Comment by anwaramoon** [comment]
> According to the unit tests I got it working now. So I'm very positive to have a release soon.
> I'd like to do some manual testing before releasing it.

Hi @astaerk is there any update when you likely to push v2.2 on nuget? I can see release tag but cannot find this package on nuget yet. Thanks

**Comment by rfcdejong** [comment]
> > According to the unit tests I got it working now. So I'm very positive to have a release soon.
> > I'd like to do some manual testing before releasing it.
> 
> Hi @astaerk is there any update when you likely to push v2.2 on nuget? I can see release tag but cannot find this package on nuget yet. Thanks

@astaerk same here, Latest stable 2.1.0 on Nuget while we bump into the same issue

**Comment by rfcdejong** [comment]
I took the sources from 2.2.0 and I had to upgrade to Unity 5.11.7 else the .UsingLifetime<TransientLifetimeManager> was throwing a compilation error.

**Comment by potatoestl** [comment]
Hello @astaerk, I wanted to know if there was any reason version 2.2.0 was begin withheld.  I pulled the latest from master and extensively tested using latest from this repo and Unity nuget packages and found no issues with my project.

**Comment by astaerk** [comment]
Hi, I never released the version because I had (and still have) trouble upgrading my biggest project to a newer Unity version. There is a change in behavior in Unity regarding open generics, but I can't pin it down.
I was debugging Unity back at the time, but there were too many radical changes in Unity (deleted classes that were replace by new concepts) to wrap my head around it. So I was never sure if Unity really had a breaking change or if autoregistration had to do something different to make it work again.

Well, and then life happened and I stopped software development in my spare time almost completely.

Since it seems to work for you guys I just released version 2.2.1 with support for Unity 5.9.

When I run the unittests against newer Unity versions this seems to work for Unity 5.9 and 5.10.

But for 5.11 it breaks the tests. I created a new issue for that: #9
